src/main/java/main/java/Calculator.java

```diff
@@ -0,0 +1,74 @@
+package main.java;
+
+import java.util.*;
+
+public class Calculator {
+    static int calculate(String e) {
+        if(e == null || e.length() < 1 || !Character.isDigit(e.charAt(0)) || !e.matches("[0-9|x%+!-]+"))
```

Comment:
You could make this more readable:
```java
boolean isFirstDigitNumber= e != null && e.length() > 0 && Character.isDigit(e.charAt(0));
if(!isFirstDigitNumber) return -1;

boolean isValidExpression = e.matches("[0-9|x%+!-]+");
if(!isValidExpression) return -1;
```
The problem with what you have is that it requires quite some java understanding to process some parts of it and even if you do understand the code, the line is a complex join of 4 conditions. Really hard to understand all in one go. Much easier when it's split into smaller steps.